app/extensions.js

```diff
@@ -300,6 +331,12 @@ module.exports.init = () => {
   process.on('extension-unloaded', (extensionId) => {
     extensionInfo.setState(extensionId, extensionStates.DISABLED)
     extensionActions.extensionDisabled(extensionId)
+    if (extensionId === config.syncExtensionId) {
+      // Reload sync extension to restart the background script
+      setImmediate(() => {
+        enableExtension(config.syncExtensionId)
```

Comment:
you can disable it functionally, but you can't disable the extension entirely; the latter isn't necessary afaict.